### ISSUE TITLE
Add "badge" with # of streams with messages

### ISSUE THREAD

**Issue by ewoutkramer** [zulip/zulip-desktop#305]
In my workflow, I have disabled desktop notifications since we have some high-volume streams. Of course, this means I do not know whether anything new has come in.  When I close the desktop app main window, I would love for the tray icon to show whether any new messages have come in in the same way Skype works:  show a number with the number of streams with unread messages.

(this is on Windows 10 by the way)

**Comment by timabbott** [comment]
I can see something like this being useful for some workflows.  If we wanted to make that work, we'd need to expose the data from the webapp codebase to the Electron app; it's all there, so shouldn't be hard, but will require some plumbing.

That said, I'm curious: @ewoutkramer have you thought  about muting the high-traffic streams?

**Comment by ewoutkramer** [comment]
If I understand your suggestion correctly, I should mute all high-traffic streams and re-enable desktop notifications.  That's an improvement - but it won't help me if I was away from the desktop and a message came in.  It would actually improve the features: if you would just count the number of non-muted streams with new messages for display in the badge, I would be totally happy.

Discussing this, I realize my problem is really this:  I have some high-traffic streams that I monitor periodically - so I really open up zulip to read them now and then. But there are others that I'd like to react on as quickly as possible.  I could now divide these two groups by muting all streams except the "urgent" ones - the badge count would tell me that something happened in the urgent streams, even if I missed the desktop notification

**Comment by akashnimare** [comment]
@ewoutkramer we have taskbar icon and system tray icon which shows the no. of unread messages even if you disable desktop notification.
- Taskbar icon -

![image](https://user-images.githubusercontent.com/2263909/34909386-c8db5054-f8c5-11e7-84f5-708d897f9e9e.png)

- Tray icon -

![image](https://user-images.githubusercontent.com/2263909/34909398-0b72bed4-f8c6-11e7-82ea-6ab3da3e4e9c.png)

Also, there is a "Flash Taskbar on new message" setting which flashes the taskbar icon on an incoming message.

<img src="https://user-images.githubusercontent.com/26190934/28449543-207b3e5c-6de1-11e7-98e6-702290cc650d.gif"/>

I'm not sure what else we should add here 😕

**Comment by ewoutkramer** [comment]
My number of unread messages is practically always 99+ - caused by a few streams that have a huge amount of traffic.  These are not the most important streams to watch however and I read them periodically, but it means that the message count is useless for me and the "new message" setting would not help me either.

Now, a suggestion was to simply mute these high-traffic streams. This is certainly an option.

Now that I have more experience with the client, I think I know what I would like:  to be able to have the message count show just the count for the streams that I pinned to "top".  I could leave the streams that I watch now and then below - not influencing my message count, and be able to decide for myself which streams were important enough to keep bugging me about!  Maybe I would even re-enable desktop notifications again if these were limited to my favorite ones on top!

**Comment by akashnimare** [comment]
@ewoutkramer if this is your situation then I'm sure the unread count will be useless even if it only shows the no. of unread message count of the top pinned stream (since as you mentioned those streams have heavy traffic i.e. always 99+) 😄  

> to be able to have the message count show just the count for the streams that I pinned to "top"

This is not possible with the desktop app and for this, we need to make changes in the web-app. For now, the best option is -
* Mute the streams which are not important to you 
* Pinned useful streams to top
* Go to each pinned streams and enable visual desktop notification for the same 

That way you'll only get the notifications from important streams. Although the badge count will still show the unread count of all unmuted streams.

**Comment by priyank-p** [comment]
I might look into this, i think we can use preload script and detect unread count per stream and we can provide setting to display count for specific stream i think but i am not sure.

(Thankfully zulip had already done work, so we don't have to in most cases like Notifications)

**Comment by akashnimare** [comment]
Yeah, that option is available though not sure how feasible it would be as we'll need to fetch the unread counts every now and then (like every 5 sec or so from pinned streams).

**Comment by ewoutkramer** [comment]
Thanks for looking into this, guys!  I know it won't bring world peace, but would improve my workflow. I'll selectively re-enable notifications for my "top" streams as suggested above.

**Comment by akashnimare** [comment]
@timabbott this is fixed in https://github.com/zulip/zulip/pull/12677, right?

**Comment by ewoutkramer** [comment]
If I understand it correctly, #12677 would restrict the count to private msgs/mentions, not the pinned channels.  Anyway, that's an improvement that I would certainly use!
